doc/debugger.rst

```diff
@@ -922,6 +935,8 @@ Error codes
 +--------+------------------------------------------------------------------+
 | 0x03   | Not found (e.g. invalid breakpoint index)                        |
 +--------+------------------------------------------------------------------+
+| 0x04   | AppRequest error (i.e. request callback returned -1)             |
```

Comment:
The error codes are relatively high level and abstract right now, could this be abstracted into something that might be usable elsewhere too, for example "internal error", "user callback error", or something similar?

Just generally the error codes are intended to provide something that might have *programmatic* meaning, i.e. trigger a different behavior in the debug client. That's why "not found" is separate because it can be used for feature detection. For most errors a debug client would simply display an error string in a log and/or an error prompt, so they don't need fine grained distinction.

Does this need to be distinguished programmatically or would 0x00 (unspecified error) + error string be enough?